xapian-applications/omega/handler_libebook.cc

```diff
@@ -0,0 +1,133 @@
+/** @file handler_libebook.cc
+ * @brief Extract text and metadata using libe-book.
+ */
+/* Copyright (C) 2019 Bruno Baruffaldi
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "handler.h"
+
+#include <xapian.h>
+#include <string.h>
+#include <unordered_map>
+#include <memory>
+#include <iostream>
+
+#include <librevenge-0.0/librevenge-generators/librevenge-generators.h>
+#include <librevenge-0.0/librevenge-stream/librevenge-stream.h>
+#include <libe-book/libe-book.h>
+
+using libebook::EBOOKDocument;
+using namespace std;
+using namespace librevenge;
+
+static
+char* str_copy(const char* text) {
+    int len = strlen(text);
+    char* line = new char[len];
+    strcpy(line, text);
+    return line;
+}
+
+static
+void clear_text(string& str,const char* text) {
```

Comment:
And similarly here, plus there's a missing space after the `,` (which `xapian-check-patch` isn't spotting for some reason).